arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/FindClosestFacilityInteractiveViewController.swift

```diff
@@ -0,0 +1,125 @@
+            AGSFacility(point: AGSPoint(x: -1.3039784633928463E7, y: 3856692.5980474586, spatialReference: .webMercator())),
+            AGSFacility(point: AGSPoint(x: -1.3049023883956768E7, y: 3861993.789732541, spatialReference: .webMercator()))]
+        let facilitySymbol = AGSPictureMarkerSymbol(url: facilityURL)
+        facilitySymbol.height = 30
+        facilitySymbol.width = 30
+        
+        for eachFacility in facilities {
+            facilityGraphicsOverlay.graphics.add(AGSGraphic(geometry: eachFacility.geometry, symbol: facilitySymbol, attributes: .none))
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["FindClosestFacilityInteractiveViewController"]
+    }
+}
+
+// MARK: - AGSGeoViewTouchDelegate
+extension FindClosestFacilityInteractiveViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Find the closest facilities with the default parameters.
+        closestFacilityTask.defaultClosestFacilityParameters { [weak self] (parameters, error) in
+            guard let self = self else { return }
+            if let parameters = parameters {
+                parameters.setFacilities(self.facilities)
+                
+                // Create a symbol for the incident.
+                let incidentSymbol = AGSSimpleMarkerSymbol(style: .cross, color: .black, size: 20)
+                
+                // Remove previous graphics.
+                self.incidentGraphicsOverlay.graphics.removeAllObjects()
+                
+                // Create a point and graphics for the incident.
+                let incidentPoint = AGSPoint(x: mapPoint.x, y: mapPoint.y, spatialReference: .webMercator())
+                let graphic = AGSGraphic(geometry: incidentPoint, symbol: incidentSymbol, attributes: .none)
+                self.incidentGraphicsOverlay.graphics.add(graphic)
+                
+                // Set the incident for the parameters.
+                parameters.setIncidents([AGSIncident(point: incidentPoint)])
+                self.closestFacilityTask.solveClosestFacility(with: parameters) { [weak self] (result, error) in
+                    guard let self = self else { return }
+                    if let result = result {
+                        
+                        // Get the ranked list of colsest facilities.
+                        let rankedList = result.rankedFacilityIndexes(forIncidentIndex: 0)
+                        
+                        // Get the facility closest to the incident.
+                        let closestFacility = Int(truncating: (rankedList?[0])!)
```

Comment:
`NSNumber` conversion is typically done through casting:

```suggestion
                        let closestFacility = rankedList?.first as! Int
```